main.tf

```
@@ -675,172 +259,172 @@ resource "aws_iam_user" "s3_bucket_iam_user" {
 
   name        = "${var.iam_user_policy_name}-S3BucketKMSPolicy"
   policy      = data.aws_iam_policy_document.s3_bucket_with_kms_policy_document_2[0].json
   description = "Policy for KMS permissions when a KMS alias is specified"
 }
 
 resource "aws_iam_user_policy_attachment" "attach_s3_bucket_with_kms_iam_policy_2" {
-  count = var.kms_alias != "" && length(var.whitelist_ip) == 0 && length(var.whitelist_vpc) == 0 && var.website_hosting == "false" ? var.number_of_users : 0
+  count = local.use_kms_encryption && length(var.whitelist_ip) == 0 && length(var.whitelist_vpc) == 0 ? var.number_of_users : 0
 
   user       = aws_iam_user.s3_bucket_iam_user[count.index].name
   policy_arn = aws_iam_policy.s3_bucket_with_kms_iam_policy_2[0].arn
 }
 
 resource "aws_iam_policy" "s3_bucket_with_kms_and_whitelist_iam_policy_1" {
-  count = var.kms_alias != "" && length(var.whitelist_ip) != 0 && length(var.whitelist_vpc) == 0 && var.website_hosting == "false" ? 1 : 0
+  count = local.use_kms_encryption && length(var.whitelist_ip) != 0 && length(var.whitelist_vpc) == 0 ? 1 : 0
 
   name        = "${var.iam_user_policy_name}-WhitelistedS3BucketObjectPolicy"
   policy      = data.aws_iam_policy_document.s3_bucket_with_kms_policy_document_whitelist_1[0].json
   description = "Policy for bucket and object permissions when a KMS alias and whitelist IP range is specified"
 }
 
 resource "aws_iam_user_policy_attachment" "attach_s3_bucket_with_kms_and_whitelist_iam_policy_1" {
-  count = var.kms_alias != "" && length(var.whitelist_ip) != 0 && length(var.whitelist_vpc) == 0 && var.website_hosting == "false" ? var.number_of_users : 0
+  count = local.use_kms_encryption && length(var.whitelist_ip) != 0 && length(var.whitelist_vpc) == 0 ? var.number_of_users : 0
 
   user       = aws_iam_user.s3_bucket_iam_user[count.index].name
   policy_arn = aws_iam_policy.s3_bucket_with_kms_and_whitelist_iam_policy_1[0].arn
 }
 
 resource "aws_iam_policy" "s3_bucket_with_kms_and_whitelist_iam_policy_2" {
-  count = var.kms_alias != "" && length(var.whitelist_ip) != 0 && length(var.whitelist_vpc) == 0 && var.website_hosting == "false" ? 1 : 0
+  count = local.use_kms_encryption && length(var.whitelist_ip) != 0 && length(var.whitelist_vpc) == 0 ? 1 : 0
 
   name        = "${var.iam_user_policy_name}-WhitelistedS3BucketKMSPolicy"
   policy      = data.aws_iam_policy_document.s3_bucket_with_kms_policy_document_whitelist_2[0].json
   description = "Policy for KMS permissions when a KMS alias and whitelist IP range is specified"
 }
 
 resource "aws_iam_user_policy_attachment" "attach_s3_bucket_with_kms_and_whitelist_iam_policy_2" {
-  count = var.kms_alias != "" && length(var.whitelist_ip) != 0 && length(var.whitelist_vpc) == 0 && var.website_hosting == "false" ? var.number_of_users : 0
+  count = local.use_kms_encryption && length(var.whitelist_ip) != 0 && length(var.whitelist_vpc) ? var.number_of_users : 0
```

```suggestion
  count = local.use_kms_encryption && length(var.whitelist_ip) != 0 && length(var.whitelist_vpc) == 0 ? var.number_of_users : 0
```

Found via the Drone build failing (also lines 352, 367, 382 and 397)

---

hmm, sorry about that. I guess my macro went wrong.
Fixed and double-checked the changes compared to the previous commit to make sure that `== 0` or `!= 0` was restored as it should have been